packages/tenant-process/src/services/tenantService.ts

```diff
@@ -1189,6 +1190,100 @@ export function tenantServiceBuilder(
       }
       return tenant.data;
     },
+
+    async m2mRevokeCertifiedAttribute({
+      organizationId,
+      tenantOrigin,
+      tenantExternalId,
+      attributeExternalId,
+      correlationId,
+      logger,
+    }: {
+      organizationId: TenantId;
+      tenantOrigin: string;
+      tenantExternalId: string;
+      attributeExternalId: string;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<void> {
+      logger.info(`Revoking certified attribute to tenant`);
+      const requesterTenant = await retrieveTenant(
+        organizationId,
+        readModelService
+      );
+
+      const certifierId = requesterTenant.data.features.find(
+        (f) => f.type === "PersistentCertifier"
+      )?.certifierId;
+
+      if (!certifierId) {
+        throw tenantIsNotACertifier(requesterTenant.data.id);
+      }
+      const targetTenant = await retrieveTenantByExternalId({
+        tenantOrigin,
+        tenantExternalId,
+        readModelService,
+      });
+
+      const attributeToRevoke = await retrieveCertifiedAttribute({
+        attributeOrigin: certifierId,
+        attributeExternalId,
+        readModelService,
+      });
+
+      const attributePreviouslyAssigned = targetTenant.data.attributes.find(
+        (attr): attr is CertifiedTenantAttribute =>
+          attr.type === tenantAttributeType.CERTIFIED &&
+          attr.id === attributeToRevoke.id
+      );
+
+      if (!attributePreviouslyAssigned) {
+        throw attributeNotFoundInTenant(
+          attributeToRevoke.id,
+          targetTenant.data.id
+        );
+      }
+
+      const tenantWithAttributeRevoked = await revokeCertifiedAttribute(
+        targetTenant.data,
+        attributeToRevoke.id
+      );
+
+      const attributeAssignmentEvent =
+        toCreateEventTenantCertifiedAttributeRevoked(
+          targetTenant.metadata.version,
+          tenantWithAttributeRevoked,
+          attributeToRevoke.id,
+          correlationId
+        );
+
+      const updatedTenantKind = await getTenantKindLoadingCertifiedAttributes(
+        readModelService,
+        tenantWithAttributeRevoked.attributes,
+        tenantWithAttributeRevoked.externalId
+      );
+
+      if (updatedTenantKind !== tenantWithAttributeRevoked.kind) {
+        const tenantWithUpdatedKind: Tenant = {
+          ...tenantWithAttributeRevoked,
+          kind: updatedTenantKind,
+        };
+        const tenantKindUpdatedEvent = toCreateEventTenantKindUpdated(
+          targetTenant.metadata.version + 1,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          tenantWithAttributeRevoked.kind!,
```

Comment:
after some considerations this behaviour is no more correct because the kind may be empty (when the tenant is created).
@rGregnanin this needs to be updated as well in the PR that it will be opened for the `kind` fix.